src/shogun/io/serialization/JsonSerializer.cpp

```diff
@@ -173,6 +173,11 @@ class JSONWriterVisitor : public AnyVisitor
 		SG_DEBUG("writing std::string with value {}", v->c_str());
 		m_json_writer.String(v->c_str());
 	}
+	void on(AutoValueEmpty* v) override
+	{
+		SG_DEBUG("writing empty auto value");
+		m_json_writer.String("auto");
```

Comment:
@karlnapf this is unlikely to happen, but a user might serialize an object which has not been fitted, so need to handle the default value of automatic parameter instantiation

Reply:
mmmmh....I guess auto can appear in many places so it is just about justifiable? Not sure....

Reply:
I guess yes :)

Reply:
The other question is what happens with get? In python I just return the value if it has been computed or the string "auto" if not. But in C++ this is not possible, unless we return a variant.. And will have issues with the other interfaces.. I was thinking if maybe it should return NaN if the auto value has not been set, or maybe just an error?

Reply:
This is what python looks like right now
```python
In [1]: import shogun as sg                                                                                                                                                                                 

In [2]: k = sg.create_kernel("GaussianKernel")                                                                                                                                                              

In [3]: k                                                                                                                                                                                                   
Out[3]: GaussianKernel(num_subscriptions={function}, cache_size=10, lhs=null, rhs=null, lhs_equals_rhs=false, num_lhs=0, num_rhs=0, combined_kernel_weight=1, optimization_initialized=false, properties=0, normalizer=IdentityKernelNormalizer(...), opt_type=FASTBUTMEMHUNGRY, m_distance=EuclideanDistance(...), m_precomputed_distance=null, log_width=auto)

In [4]: k.get("log_width")                                                                                                                                                                                  
Out[4]: 'auto'
```

Reply:
I think I'd prefer an error, unified for all interfaces... only the to_string method should print the auto ...

Reply:
The alternative would be to return nan, but I dont like that too much I think

Reply:
> I think I'd prefer an error, unified for all interfaces... only the to_string method should print the auto ...

Cool, makes my life easier :D

> The alternative would be to return nan, but I dont like that too much I think

Yea the issue with that is that there is no standard for non floating point representations of nan. And I don't even know if all languages we support have implemented NaN according to the ISO standard